pallets/phala/src/fat.rs

```diff
@@ -125,12 +199,25 @@ pub mod pallet {
 			cluster_id: ContractClusterId,
 		) -> DispatchResult {
 			let origin: T::AccountId = ensure_signed(origin)?;
-			// TODO.shelven: check permission?
-			Self::push_message(ContractOperation::UploadCodeToCluster {
-				origin,
-				code,
-				cluster_id,
-			});
+			let cluster_info = Clusters::<T>::get(cluster_id).ok_or(Error::<T>::ClusterNotFound)?;
+			// only allow code upload when all the workers deploy the cluster
+			// the same check is not needed in instantiate_contract() any more
+			let deployed_workers = ClusterWorkers::<T>::get(cluster_id);
+			ensure!(
+				cluster_info.workers == deployed_workers,
+				Error::<T>::ClusterPermissionDenied
```

Comment:
More explicit error?

Reply:
Remove this check since we share `ClusterKey` to all workers in one message now.